mlflow/store/artifact/databricks_artifact_repo.py

```
@@ -398,6 +406,107 @@ def _get_run_relative_artifact_path_for_upload(self, src_file_path, dst_artifact
+                    part_etags.append(PartEtag(part_number=part_number, etag=etag))
+
+        return part_etags
+
+    def _complete_multipart_upload(self, run_id, path, upload_id, part_etags):
+        return self._call_endpoint(
+            DatabricksMlflowArtifactsService,
+            CompleteMultipartUpload,
+            message_to_json(
+                CompleteMultipartUpload(
+                    run_id=run_id,
+                    path=path,
+                    upload_id=upload_id,
+                    part_etags=part_etags,
+                )
+            ),
+        )
+
+    def _abort_multipart_upload(self, cred_info):
+        headers = self._extract_headers_from_credentials(cred_info.headers)
+        with rest_utils.cloud_storage_http_request(
+            "delete", cred_info.signed_uri, headers=headers
+        ) as response:
+            augmented_raise_for_status(response)
+            return response
+
+    def _multipart_upload(self, local_file, artifact_path):
+        num_parts = math.ceil(os.path.getsize(local_file) / _MULTIPART_UPLOAD_CHUNK_SIZE)
+        create_mpu_resp = self._create_multipart_upload(self.run_id, artifact_path, num_parts)
+        try:
+            part_etags = self._upload_parts(
+                local_file,
+                self.run_id,
+                artifact_path,
+                create_mpu_resp.upload_id,
+                create_mpu_resp.upload_credential_infos,
+            )
+            self._complete_multipart_upload(
+                self.run_id,
+                artifact_path,
+                create_mpu_resp.upload_id,
+                part_etags,
+            )
+        except Exception as e:
+            _logger.warning(
+                "Encountered an unexpected error during multipart upload: %s, aborting. "
+                "Set logging level to DEBUG to see the full traceback.",
+                e,
+            )
+            _logger.debug("", exc_info=True)
```

If there's an unexpected exception during multipart upload, we should throw the exception. Otherwise, the user's log_artifact() call won't throw but there won't be any file logged. This could be confusing. Can we rethrow after aborting and then add a test case to ensure we throw?